hydrabus/hydrabus_bbio_spi.c

```diff
@@ -127,13 +138,15 @@ void bbio_mode_spi(t_hydra_console *con)
 					cprint(con, "\x00", 1);
 					break;
 				}
-				chnRead(con->sdu, tx_data, to_tx);
+				if(to_rx > 0) {
+					chnRead(con->sdu, tx_data, to_tx);
 
-				if(bbio_subcommand == BBIO_SPI_WRITE_READ) {
-					bsp_spi_select(proto->dev_num);
+					if(bbio_subcommand == BBIO_SPI_WRITE_READ) {
```

Comment:
The checking shouldn't be inside to if(to_tx > 0) block, otherwise the chip would never be selected if the length is 0